src/createStaticPackage.js

```diff
@@ -21,15 +23,56 @@ const IFRAME_CONTENT = `
 </html>
 `;
 
+/**
+ * Gets all files in a directory and all of its subdirectories
+ *
+ * The returned value is sorted for deterministic output.
+ *
+ * @param {string} dir
+ * @returns {Array<string>}
+ */
+function getFilesRecursive(dir) {
+  const files = [];
+  const dirs = [dir];
+
+  while (dirs.length > 0) {
+    const currentDir = dirs.shift();
+    // Node 18 adds `recursive` option to `fs.readdirSync`, which would
+    // make this function simpler.
+    const dirents = fs.readdirSync(currentDir, { withFileTypes: true });
+
+    for (const dirent of dirents) {
+      const res = path.resolve(currentDir, dirent.name);
+
+      if (dirent.isDirectory()) {
+        // This is a directory, so we are going to add it to the list of directories to recurse into.
+        dirs.push(res);
+        files.push(res);
+      } else {
+        files.push(res);
+      }
+    }
+  }
+
+  // Sort the files for deterministic output. This is important so that
+  // the archive hash is the same.
+  files.sort((a, b) => a.localeCompare(b));
+  return files;
+}
+
 export default function createStaticPackage({ tmpdir, publicFolders }) {
   return new Promise((resolve, reject) => {
-    const archive = new Archiver('zip');
+    const archive = new Archiver('zip', {
+      // Concurrency in the stat queue leads to non-deterministic output.
+      // https://github.com/archiverjs/node-archiver/issues/383#issuecomment-2253139948
+      statConcurrency: 1,
```

Comment:
As far as I can tell, this is the most important part of the change. The rest of the code changes here address possible ordering inconsistencies in the globbing, which at this point are only theoretical.